src/Init/Data/Repr.lean

```diff
@@ -347,6 +347,7 @@ def Char.quoteCore (c : Char) : String :=
   else if  c = '\t' then "\\t"
   else if  c = '\\' then "\\\\"
   else if  c = '\"' then "\\\""
+  else if  c = '\'' then "\\\'"
```

Comment:
If you add `(inString : Bool := false)` to the signature, then you can do
```suggestion
  else if inString && c = '\'' then "\\\'"
```
and then later have
```lean
def String.quote (s : String) : String :=
  if s.isEmpty then "\"\""
  else s.foldl (fun s c => s ++ c.quoteCore (inString := true)) "\"" ++ "\""
```